src/core_plugins/timelion/public/directives/timelion_expression_input_helpers.js

```diff
@@ -52,13 +69,69 @@ export class FunctionSuggestions {
   }
 }
 
-export function suggest(expression, functionList, Parser) {
+function inLocation(cursorPosition, location) {
+  return cursorPosition >= location.min && cursorPosition <= location.max;
+}
+
+function extractSuggestionsFromParsed(result, cursorPosition, functionList) {
+  const activeFunc = result.functions.find((func) => {
+    return cursorPosition >= func.location.min && cursorPosition < func.location.max;
+  });
+
+  if (activeFunc) {
```

Comment:
How about incorporating an early exit, and allowing this branch to be the default?

```javascript
if (!activeFunc) {
  return;
}

const funcDefinition = functionList.find(...
```